daemon/network.go

```diff
@@ -99,15 +100,51 @@ func (daemon *Daemon) getAllNetworks() []libnetwork.Network {
 	return daemon.netController.Networks()
 }
 
-func isIngressNetwork(name string) bool {
-	return name == "ingress"
+type ingressJob struct {
+	create *clustertypes.NetworkCreateRequest
+	ip     net.IP
 }
 
-var ingressChan = make(chan struct{}, 1)
+var (
+	ingressWorkerOnce  sync.Once
+	ingressJobsChannel chan *ingressJob
+	ingressWorkerStop  chan struct{}
+	ingressID          string
+	ingress            sync.Mutex
```

Comment:
@aaronlehmann I decided to fall back to not stopping the routine, when the node leaves the swarm.

IMO it is fine for now as we are adding the new functionality.
We can later improve the logic getting to ideal stop of the routine, with calm.

I updated the diffs and tested the add/rm ingress, before and after leaving and rejoining the cluster.